### ISSUE TITLE
Console: font settings not working

### ISSUE THREAD

**Issue by GeVanCo** [gitextensions/gitextensions#10933]
### Environment

- Git Extensions 4.0.2.16100
- Build 25100ec1f7c8da8f798613da74826038af81a50e
- Git 2.39.0.windows.2
- Microsoft Windows NT 10.0.19044.0
- .NET 6.0.16
- DPI 96dpi (no scaling)
- Microsoft.WindowsDesktop.App Versions

```
    Microsoft.WindowsDesktop.App 3.1.25 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

### Issue description

I want to change the font size of the Console tab in Git Extensions 4.0.2.  I'm using xterm as console style.

When I go to Tools -> Settings  -> Git Extensions -> Appearance -> Console Style and change the font size (which defaults to Consolas, 10) to whatever value, the size of the font in the Console tab doesn't change after a restart.

### Steps to reproduce

Go to the Console Style section of the Settings, select xterm as console style and change the default font size to another value.  Then restart Git Extensions and go to the Console tab: font size is not changed.

### Did this work in previous version of GitExtensions?

Yes, versions 3 of Git Extensions

**Comment by GeVanCo** [comment]
Also here, I tried with an older version of Git Extensions with the following info:

- Git Extensions 3.5.1.12196
- Build 9099a1e76d01dbbd6896d6b026027db0b68fba96
- Git 2.39.0.windows.2
- Microsoft Windows NT 10.0.19044.0
- .NET Framework 4.8.4614.0
- DPI 96dpi (no scaling)

And when I change the font in the Console settings to a (yes, I know, absurd) value of 20 then I do see the following result:

![font](https://user-images.githubusercontent.com/3616771/235424353-d74b864e-6187-4f1f-83a6-e085341e3abd.png)

Result:

![font_ok](https://user-images.githubusercontent.com/3616771/235424421-8f840cf7-60cd-4390-9742-3d0bcdcfc9b5.png)

So, yes, also here it was working with older versions of Git Extension.

Best,
--Geert

**Comment by mstv** [comment]
Only the chosen font size seems to be ignored. The font style is applied.
both in 4.0.2 and 4.1 beta

**Comment by GeVanCo** [comment]
What I noticed: whether I select _Default Windows Scheme_ or _xterm_ for my console type, they're exactly the same.  I remember from previous versions Xterm had a distinct view compared to all the other views?  The command line was coloured and so on.  Nothing of this here.

Default Windows Scheme:

![default_windows](https://user-images.githubusercontent.com/3616771/235475169-73b6cd6a-cb5d-40e8-b2c3-d035794f02b8.png)

xterm:

![xterm_scheme](https://user-images.githubusercontent.com/3616771/235475325-0918e26b-87dc-4fdc-8398-e044d0583089.png)

I don't see any differences and also: why do I see Microsoft stuff in the initial xterm view?  I think the whole xterm view is broken...

**Comment by mstv** [comment]
The .NET `Font.Size` is of type `float`. A font size displayed as 20 is stored as 20.25 f.i.
Since ConEmu expects an integer value, the `ConEmuControl` tries to convert it and silently fails. (I hate error hiding!)
Will be improved by #10938.

---

> why do I see Microsoft stuff in the initial xterm view

This was the right question. Because you have configured "cmd" as shell.
If you start a bash, you get the colors...
(only in the console tab, but not in the progress window)

**Comment by GeVanCo** [comment]
> > why do I see Microsoft stuff in the initial xterm view
> 
> This was the right question. Because you have configured "cmd" as shell. If you start a bash, you get the colors... (only in the console tab and only after GE restart, but not in the progress window)

Well, no...  I **do** have "bash" selected as default shell in the Detailed -> Browse repository window, **not** the cmd shell.  And still, xterm shows me MS-alike views.

**Comment by mstv** [comment]
> when I select bash or cmd: they're looking and **behaving** exactly the same.

So there seems to be something in your system which uses `cmd.exe` as a replacement of `bash.exe` (have a look at the prompt in your screenshots).

You can specify the folder of `bash.exe`:

![image](https://user-images.githubusercontent.com/36601201/236668190-7d336006-8afc-4f80-a8a9-fccba405955b.png)

**Comment by GeVanCo** [comment]
@mstv : your suggestion triggered me.

First off, the path to the Linux tools was not set.  Resolving this gave me back my "real" Bash/Xterm behaviour (that is, coloured command prompt) like so:

![wipi](https://user-images.githubusercontent.com/3616771/236671139-47057ebd-e5d3-4059-bd35-2967cbcb0b30.png)

Next, I still had nothing when executing whatever git command (even typing git + return was giving me nothing back).  Then I ran the command "which git" and saw it was pointing to a git exe somewhere in the Cygwin environment.  I changed the path to the ...../Git/cmd/git.exe and also made sure this path was set before the Cygwin path in my PATH env. variable on Windows.

That resolved everything related to the misbehaviour of the GE console window.  I have now this (including the git branch I'm on):

![gitok](https://user-images.githubusercontent.com/3616771/236671285-d9b6b383-612a-4f28-a77f-f93b83d49a01.png)

Thanks for this suggestion, it put me on the right track!

PS: only thing left is the font size but I've seen that will be fixed in the upcoming version 4.1

Again, thanks!

**Comment by gluttony38** [comment]
Thanks @mstv, got a similar issue with weird behavior (missing environment variables, commands not acting as expected, ...) in Git Extensions' console, it seems some WSL or another Windows Linux tools install or upgrade changed PATH setting another Linux tool, forcing it to the one from Git fixed the issue \o/ !
